mobile-app/app/screens/AppNavigator/screens/Loans/hooks/LoanPaymentTokenRate.ts

```diff
@@ -1,69 +1,100 @@
 import { useSelector } from 'react-redux'
 import BigNumber from 'bignumber.js'
-import { LoanToken } from '@defichain/whale-api-client/dist/api/loan'
 import { RootState } from '@store'
-import { loanPaymentTokenActivePrice } from '@store/loans'
+import { loanPaymentTokenActivePrices } from '@store/loans'
 import { getActivePrice } from '../../Auctions/helpers/ActivePrice'
-import { PaymentTokenProps } from '../screens/PaybackLoanScreen'
+import { tokenSelectorByDisplaySymbol, tokensSelector, WalletToken } from '@store/wallet'
+import { TokenData } from '@defichain/whale-api-client/dist/api/tokens'
+
+export interface PaymentTokenProps {
+  tokenId: string
+  tokenSymbol: string
+  tokenDisplaySymbol: string
+  tokenBalance: BigNumber
+}
+
 interface GetAmountProps {
   paymentTokenAmounts: Array<{
-    resultingBalance: BigNumber
-    amountToPayInPaymentToken: BigNumber
     amountToPayInLoanToken: BigNumber
+    amountToPayInPaymentToken: BigNumber
+    cappedAmount: BigNumber
+    outstandingBalanceInPaymentToken: BigNumber
     paymentToken: PaymentTokenProps
   }>
 }
 
 export const useLoanPaymentTokenRate = (props: {
   loanTokenAmountActivePriceInUSD: BigNumber
-  paymentTokens: PaymentTokenProps[]
-  loanToken: LoanToken | undefined
+  loanToken: {
+    id: string
+    displaySymbol: string
+    symbol: string
+  }
   outstandingBalance: BigNumber
   amountToPay: BigNumber
   loanTokenBalance: BigNumber
 }): {
-  getAmounts: () => GetAmountProps
+  getPaymentTokens: () => GetAmountProps
+  getPaymentPenalty: (paymentTokenSymbol: string) => BigNumber
 } => {
-  const paymentTokenActivePrice = useSelector((state: RootState) => loanPaymentTokenActivePrice(state.loans)) // DFI
-  const getAmounts = (): GetAmountProps => {
-    const paymentTokenAmounts = props.paymentTokens.map(paymentToken => {
-      const paymentTokenActivePriceInUSD = getActivePrice(paymentToken.tokenSymbol ?? '', paymentTokenActivePrice) // DUSD, dTU10
-      const conversionRate = paymentToken.tokenSymbol === props.loanToken?.token.symbol
+  const tokens = useSelector((state: RootState) => tokensSelector(state.wallet))
+  const DUSDToken = useSelector((state: RootState) => tokenSelectorByDisplaySymbol(state.wallet, 'DUSD'))
+  const paymentTokenActivePrices = useSelector((state: RootState) => loanPaymentTokenActivePrices(state.loans))
+  const paymentTokens = _getPaymentTokens({
+    ...props.loanToken
+  }, props.loanTokenBalance, tokens, DUSDToken)
+
+  const getPaymentPenalty = (paymentTokenSymbol: string): BigNumber => {
```

Comment:
Should we move this out of the hook? As it will be reinitialized everytime. Maybe better to make it a pure function